client/js/chart/index.js

```diff
@@ -126,6 +116,34 @@ export function createChart(el, rankingHistory, rankingDates) {
   el.append(tooltip);
 }
 
+/**
+ * Tests whether the mouse is currently near a point on the chart.
+ *
+ * @param {SVGSVGElement} svg The SVG element for the chart.
+ * @param {number[][]} rows The ranks of each series at each date.
+ * @param {number} clientX The position of the mouse.
+ * @param {number} clientY The position of the mouse.
+ * @returns {{hit: number | null, x: number | null, y:number | null}} A hit
+ * result.
+ */
+function hitTest(svg, rows, clientX, clientY) {
+  // convert mouse coords to SVG coords
+  const mousePos = svg.createSVGPoint();
+  mousePos.x = clientX;
+  mousePos.y = clientY;
+  const chartPos = mousePos.matrixTransform(svg.getScreenCTM().inverse());
+
+  // index in history
+  const x = Math.round(chartPos.x / RUN_SCALE_X);
+  // value in history
+  const y = Math.round(chartPos.y / RUN_SCALE_Y);
```

Comment:
What is `history`?

Reply:
these comments refer to the history of the song's position (on july 8th, it was ranked third, on july 9th, it was ranked second, etc)